internal/dot/graph.go

```diff
@@ -42,3 +44,29 @@ type Node struct {
 	Optional bool
 	Group    string
 }
+
+// String returns the string representation of a node so the different
+// constructors can refer to the same node. We omit information on the optional
+// field since the same type can be optional to one constructor and required
+// for another.
+func (n *Node) String() string {
+	if n.Name != "" {
+		return fmt.Sprintf("%v[name=%v]", n.Type, n.Name)
+	} else if n.Group != "" {
+		return fmt.Sprintf("%v[group=%v]", n.Type, n.Group)
+	}
+
+	return n.Type
+}
+
+// Attributes composes and returns a string to style the sublabels when
+// visualizing graph.
+func (n *Node) Attributes() string {
+	if n.Name != "" {
+		return fmt.Sprintf("<BR /><FONT POINT-SIZE=\"10\">Name: %v</FONT>", n.Name)
```

Comment:
Go provides "`" for string literals so that you don't have to escape double quotes.

    return fmt.Sprintf(`<BR /><FONT POINT-SIZE="10">Name: %v</FONT>`, n.Name)